docs/design/2020-01-24-collations.md

```diff
@@ -0,0 +1,243 @@
+  2. Users of MySQL who requires collations can move to TiDB without modifying their scripts/applications.
+  3. Users of TiDB can distinguish the tables with old collation and new collations easily, if they both exist in a TiDB cluster.
+  4. All replications/backups/recovers through TiDB tools like Binlog-related mechanism, CDC and BR should not encounter unexpected errors caused from the changes of collations in TiDB. For example, 
+    - When using BR, if backup data from old cluster is applied to the new cluster, the new cluster should know that the data is binary collated without padding, or otherwise 'Duplicate entry' error may be reported for primary key/unique key columns.
+    - If we don't allow both old collations and new collations exist in a TiDB cluster(the Option 4, see below), trying to make replications/backup & recovery between old/new clusters will break this prerequisite.
+
+#### Option 1
+
+Add a series of new collatins named with the suffix "`_np_bin`"(meaning "NO PADDING BINARY"), for example, `utf8mb4_np_bin`. Such new collations don't exist in MySQL, but they're the "real ones" used by current TiDB. After upgrading to newer TiDB versions, all old collations are shown as "`_np_bin`", MySQL collations behave the same with MySQL.
+
+Pros: Requirement 1.a, 2, 3 and 4 are met.
+
+Cons: Requirement 1.b, 1.c are not met.
+
+#### Option 2
+
+Keep all collations defined in TiDB as what they were, define a series of new collations that are actually compatible with MySQL collations. For example, we can create a new `tidb_utf8_mb4_general_ci` that is the same as `utf8mb4_general_ci` in MySQL. When TiDB users want the "real" collations, they can modify their `CREATE TABLE` statements to use the new ones.
+
+Pros: Requirement 1.a, 1.b, 2 and 3 are met.
+
+Cons: Requirement 1.c and 4 are not met, the namings are also confusing: collations that are named from MySQL are different from MySQL, but the ones named different from MySQL behave the same as MySQL.
+
+#### Option 3
+
+Add no new collation, tag old collations with syntax comment. For a upgraded TiDB cluster, add comment like "`/* AS_BINARY_COLLATION */`" for the old collations; all collations with "`/* AS_BINARY_COLLATION */`" comment are treated as the old ones.
+
+
+Pros: Requirement 1.a, 2, 3 and 4 are met.
+
+Cons: Requirement 1.b and 1.c are not met.
+
+#### Option 4
+
+Only TiDB clusters that initially boostraped with the new TiDB version are allowed to enable the new collations. For old TiDB clusters, everything remains unchanged after the upgrade.
+
+We can also provide a configuration entry for the users to choose between old/new collations when deploying a new clusters.
+
+Pros: Requirement 1 and 2 are met.
+
+Cons: Since existing TiDB cluster can't get new collations enabled, requirement 3 is eliminated; Requirement 4 is not met.
+
+## Implementation
```

Comment:
How about we move this section to the beginning of the proposal section?

Reply:
IMHO it's better to keep the alternatives firstly, and then implementation details.

Reply:
I've made the change, there will be a TODO list for progress tracking as well as the guidance for the community.